cmd/root.go

```diff
@@ -179,12 +204,34 @@ func startWorker(userInput string) <-chan error {
 
 func runDefaultCmd(_ *cobra.Command, args []string) error {
 	userInput := args[0]
-	errs := startWorker(userInput)
+	errs := startWorker(userInput, appConfig.FailOnSeverity)
 	ux := ui.Select(appConfig.CliOptions.Verbosity > 0, appConfig.Quiet)
 	return ux(errs, eventSubscription)
 }
 
-func ListLocalDockerImages(prefix string) ([]string, error) {
+// hitSeverityThreshold indicates if there are any severities >= to the max allowable severity (which is optional)
+func hitSeverityThreshold(thresholdSeverity *vulnerability.Severity, results result.Result, metadataProvider vulnerability.MetadataProvider) bool {
+	if thresholdSeverity != nil {
```

Comment:
could reduce the nesting in this method a bit:
```
if thresholdSeverity == nil {
    return true
}
```

Reply:
Agreed. Also, I think this could potentially be further simplified by having the severity variable not be a pointer type.

Reply:
👍 guard clases are best!